### ISSUE TITLE
'SessionTransaction' object has no attribute '_iterate_parents'

### ISSUE THREAD

**Issue by SysAd-WI** [zopefoundation/zope.sqlalchemy#18]
I'm getting following error. Already upgraded zope.sqlalchemy...
```
2017-03-02 15:18:30,463  ERROR    1e5d84d7146d44d7  indico.wsgi               'SessionTransaction' object has no attribute '_iterate_parents'
Traceback (most recent call last):
  File "/var/www/indicoenv/lib/python2.7/site-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/var/www/indicoenv/lib/python2.7/site-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/var/www/indicoenv/lib/python2.7/site-packages/indico-1.9.9-py2.7.egg/indico/web/assets/blueprint.py", line 47, in js_vars_global
    data = generate_global_file(config)
  File "/var/www/indicoenv/lib/python2.7/site-packages/indico-1.9.9-py2.7.egg/indico/web/assets/vars_js.py", line 33, in generate_global_file
    locations = Location.find_all() if config.getIsRoomBookingActive() else []
  File "/var/www/indicoenv/lib/python2.7/site-packages/indico-1.9.9-py2.7.egg/indico/core/db/sqlalchemy/util/models.py", line 99, in find_all
    return cls.find(*args, **kwargs).all()
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2679, in all
    return list(self)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2831, in __iter__
    return self._execute_and_instances(context)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2852, in _execute_and_instances
    close_with_result=True)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2861, in _get_bind_args
    **kw
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2843, in _connection_from_session
    conn = self.session.connection(**kw)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 966, in connection
    execution_options=execution_options)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 971, in _connection_for_bind
    engine, execution_options)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 417, in _connection_for_bind
    self.session.dispatch.after_begin(self.session, self, conn)
  File "/var/www/indicoenv/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 256, in __call__
    fn(*args, **kw)
  File "/usr/lib/python2.7/site-packages/zope/sqlalchemy/datamanager.py", line 231, in after_begin
    join_transaction(session, self.initial_state, self.transaction_manager, self.keep_session)
  File "/usr/lib/python2.7/site-packages/zope/sqlalchemy/datamanager.py", line 205, in join_transaction
    DataManager(session, initial_state, transaction_manager, keep_session=keep_session)
  File "/usr/lib/python2.7/site-packages/zope/sqlalchemy/datamanager.py", line 65, in __init__
    self.tx = session.transaction._iterate_parents()[-1]
AttributeError: 'SessionTransaction' object has no attribute '_iterate_parents'
```

Any suggestions or workaround here?

**Comment by tseaver** [comment]
Using a private method of `SessionTransaction` puts us clearly in the wrong here.  The fastest workaround I can think of would be to pin `sqlalchemy` to an older version which still has that method.

**Comment by tseaver** [comment]
Oops, this is a duplicate of #15, already fixed in PR #16.  You need to update to `zope.sqlalchemy 0.7.7` to use `sqlalchemy 1.1`.
